pyccel/parser/semantic.py

```diff
@@ -1919,8 +1919,7 @@ def _get_indexed_type(self, base, args, expr):
             else:
                 raise errors.report(f"Unknown annotation base {base}\n"+PYCCEL_RESTRICTION_TODO,
                         severity='fatal', symbol=expr)
-            rank = 1
-            if len(args) == 2 and args[1] is LiteralEllipsis():
+            if len(args) > 0:
```

Comment:
This condition is significantly more permissive than the old condition. It implies that you now correctly handle (or correctly raise an error for expressions such as `tuple[int, bool, float]`. Is this correct?

Reply:
this is now deprecated after I merged the `devel` branch into my branch